ui/component/or-dashboard-builder/src/widgets/or-image-widget.ts

```diff
@@ -0,0 +1,366 @@
+import { i18next } from "@openremote/or-translate";
+import { css, html, LitElement, TemplateResult, unsafeCSS } from "lit";
+import { customElement, property, state, query } from "lit/decorators.js";
+import { OrWidgetConfig, OrWidgetEntity } from "./or-base-widget";
+import { style } from "../style";
+import {debounce} from "lodash";
+import { SettingsPanelType, widgetSettingsStyling } from "../or-dashboard-settingspanel";
+import { InputType, OrInputChangedEvent } from "@openremote/or-mwc-components/or-mwc-input";
+
+export interface ImageWidgetConfig extends OrWidgetConfig {
+    displayName: string;
+    attributeRefs: AttributeRef[];
+    xCoordinatesMap: [number];
+    yCoordinatesMap: [number];
+    showTimestampControls: boolean;
+    imageUploaded: boolean;
+    imagePath: string;
+    assets: Asset[];
+}
+
+
+export class OrImageWidget implements OrWidgetEntity {
+
+    readonly DISPLAY_MDI_ICON: string = "file-image-marker"; // https://materialdesignicons.com;
+    readonly DISPLAY_NAME: string = "Image";
+    readonly MIN_COLUMN_WIDTH: number = 2;
+    readonly MIN_PIXEL_HEIGHT: number = 0;
+    readonly MIN_PIXEL_WIDTH: number = 0;
+
+    getDefaultConfig(widget: DashboardWidget): OrWidgetConfig {
+        return {
+            displayName: widget?.displayName,
+            attributeRefs: [],
+            xCoordinatesMap: [],
+            yCoordinatesMap: [],
+            showTimestampControls: false,
+            imageUploaded: false,
+            imagePath: "https://home3ds.com/wp-content/uploads/2018/11/PNG.png",
+            assets: []
+        } as unknown as ImageWidgetConfig;
+    }
+
+    
+
+    getSettingsHTML(widget: DashboardWidget, realm: string) {
+        return html`<or-image-widgetsettings .widget="${widget}" realm="${realm}"></or-image-widgetsettings>`;
+    }
+
+    getWidgetHTML(widget: DashboardWidget, editMode: boolean, realm: string) {
+        return html`<or-image-widget .widget="${widget}" .editMode="${editMode}" realm="${realm}" style="height: 100%; overflow: hidden;"></or-image-widget>`;
+    }
+
+// Triggered every update to double check if the specification.
+    // It will merge missing values, or you can add custom logic to process here.
+    verifyConfigSpec(widget: DashboardWidget): ImageWidgetConfig {
+        return Util.mergeObjects(this.getDefaultConfig(widget), widget.widgetConfig, false) as ImageWidgetConfig;
+    }
+}
+
+@customElement("or-image-widget")
+export class OrImageWidgetContent extends LitElement {
+    @query("#img-container")
+    private _imgSize!: HTMLElement;
+    @state()
+    private assetAttributes: [number, Attribute<any>][] = [];
+    @state()
+    private image?: HTMLInputElement;
+    @state()
+    private imageSize?: { width: number, height: number }
+    @state()
+    private loadedAssets: Asset[] = [];
+    
+    @property()
+    public editMode?: boolean;
+    @state()
+    public imageUploaded?: boolean;
```

Comment:
Make sure when merging you remove unused variables if present.
(not 100% sure if this is one of them)